rbc/omnisci_backend/numpy_ufuncs.py

```diff
@@ -9,7 +9,17 @@
 
 
 def determine_dtype(a, dtype):
-    return a.eltype if dtype is None else dtype
+    if isinstance(a, ArrayPointer):
+        return a.eltype if dtype is None else dtype
+    else:
+        return a if dtype is None else dtype
```

Comment:
`return a` is suspicious: the function ought to return type instance..

Reply:
`a` ought to be a type. I can add a check at the beginning of this function to make sure it is a type. What do you think?

Reply:
Perhaps introduce a decorator on  determine_dtype and determine_input_type that checks that the return type is numba type?

Reply:
Can you point the case where `return a` is effective?

Reply:
Yes. Look at the implementation of `np.negative(<int64>)` for instance. In this case `return a` returns `int64` since the dtype is `None`. But on `np.logical_not`, the output needs to be a boolean value, and we enforce this constraint using the dtype argument.